### PR TITLE
V0.0 dash

### PR DESCRIPTION
**Pull request by akd13**
Update developer guide with user stories and NFR

### CODE REVIEW

**Comment by mattheuslee** [comment]
Can the Windows 7 requirement be considered as under the "mainstream-os" requirement?

**Review by ChuaPingChan — COMMENTED** [review]
~~The "author" field of product survey should refer to the developers of Wunderlist.~~

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -316,20 +316,108 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
+
+|`* *` |user |recover deleted entries |retrieve them in case of accidental deletions
+
+|`* *` |user with many projects |tag entries by assignment |distinguish tasks for similar assignments
+
+|`* *` |user with many projects |view entries by assignment |distinguish different assignments
+
+|`* *` |user |search for free days in a time period (e.g. next 1 month) |use free days to slot tasks
+
+|`* *` |user |launch and minimize the application |
```
Comment (kevinLamKB) [inline docs/DeveloperGuide.adoc]: launch/minimize with a keyboard shortcut. 
missing the keyboard shortcut, right now it looks we want to be able to launch and minimize which of course we can

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -316,20 +316,108 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
+
+|`* *` |user |integrate reminders from e-mail |
+
+|`* *` |developer |have an automated test program |check program functionality after every update
+
+|`* *` |user |clear all entries existing before a certain date |keep calendar organized
+
+|`* *` |user |configure defaults (e.g. storage path) |
+
+|`* *` |user |launch and minimize the application |use it more easily
```
Comment (kevinLamKB) [inline docs/DeveloperGuide.adoc]: repeated

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -366,11 +454,18 @@ ____
...
+.  The software should work on any link:#mainstream-os[mainstream OS] as long as it has Java `1.8.0_60` or higher installed.
+.  The app should be able to hold up to 1000 entries without a noticeable sluggishness in performance for typical usage.
 .  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+.  The product should be a desktop app.
+.  The Command Line Inteface should be the primary mode of input.
+.  The software should work stand-alone and not be a plug-in to another app.
+.  The software should use text files for storage and not relational databases.
+.  The data should be stored locally in an editable text file.
+.  The code should follow the Object-oriented paradigm.
+.  The software should work on Windows 7 or later.
```
Comment (kevinLamKB) [inline docs/DeveloperGuide.adoc]: aggree with mattheus, overlaps with mainstream-os NFR

**Review by mattheuslee — CHANGES_REQUESTED** [review]
Can the changes be made first before we merge this PR?

**Comment by akd13** [comment]
Changes made, pls review and merge

**Review by ChuaPingChan — APPROVED** [review]

**Review by mattheuslee — APPROVED** [review]

**Pull request merged**